song_match/song_match.py

```diff
@@ -66,18 +64,21 @@ def __init__(self, num_players: int = 1):
     async def __init_game_loop(self) -> None:
         current_position = STARTING_POSITION
         while self._song.is_not_finished(current_position, self._game_length):
+            await self.__tap_guard(lambda: self.__play_round_transition_effect())
```

Comment:
Can you remind me why it's necessary to play this at the beginning of rounds as opposed to the end? I think I prefer it if it was at the end.

Reply:
I'd prefer to put it there too except that the is_not_finished method (that checks whether the current position has reached the last index) is at the beginning of the loop. If __play_round_transition were at the bottom then it would show the effect after the final round followed by nothing when the game ends by reaching the end of the song.

Reply:
Could you move it back to the bottom of the loop if you add a guard statement?

```python
    async def __play_round_transition_effect(self, round_number: int):
        if self.__is_not_last_round(round_number):
            effect = self._effect_factory.create('RoundTransition')
            await effect.play()
```